src/ipc/handler.c

```diff
@@ -496,7 +496,7 @@ static int ipc_dai_config(uint32_t header)
 	}
 
 	/* configure DAI */
-	ret = dai_set_config(dai, &config);
+	ret = dai_set_config(dai, _ipc->comp_data);
```

Comment:
@lgirdwood @dabekjakub This seems like a recurring theme. @keyonjie fixed a similar one a couple of days ago in https://github.com/thesofproject/sof/pull/703.
I'll check all the other handler functions too.

Reply:
Yes that is a recurring theme. This is the same issue with flex arrays. Maybe we should just allocate data from runtime heap with size of ipc itself for the config and then use ipc size for memcopy. We will have flex arrays and won't cut off the message itself since it won't be allocated on the stack.

```suggestion
	struct ipc_comp_dev *comp_dev;
	struct sof_ipc_ctrl_data *data = rmalock(RZONE_RUNTIME,SOF_MEM_CAPS_RAM, 
	((struct sof_ipc_cmd_hdr *)_ipc->comp_data)->size);
```

Reply:
@dabekjakub yes that will solve the problem but is it advisable to always allocate max ipc size? I think it might be wasteful to do that. maybe @lgirdwood can comment on it.

Reply:
I didn' say maximum :) The fragment
```code
	((struct sof_ipc_cmd_hdr *)_ipc->comp_data)->size)
```
takes ipc size from ipc header that is filled by host.

Reply:
@dabekjakub oops sorry I misunderstood. This is correct. Lets make that change incrementally.

Reply:
@ranj063 @lgirdwood I am not sure why the existing implementation makes a copy on the stack instead of processing the request in-place by all the layers? Is the copy on stack/heap necessary?